app/presenters/tool/presenter.rb

```diff
@@ -0,0 +1,13 @@
+class Tool::Presenter
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  delegate :slug, :name, :intro, :content, :meta_image_url, to: :active_record
+
+  private
+    attr_reader :active_record
+
+    def active_record
+      raise NotImplementedError, "Each Tool::Presenter must know how to fetch its Tool active record"
```

Comment:
> Is there a reason we're storing the tools in the database (and seeding them) rather than just hardcoding the information into the presenter and pointing a new route at them?

I don't see a compelling reason, just that it's the way it was built initially.

I also had the instinct to hardcode the info. I'd be +1 to refactor things that way at some point.

Persisting in the DB is the reason this comment was necessary: https://github.com/maybe-finance/marketing/pull/134#discussion_r1760164050

> The current workflow to create a new tool would be this correct?

I've updated the PR description with the steps to add a new tool. The seeds in this repo seem to have been thought out to populate the dev env, but never the prod env. Another thing I'd be +1 to change.

Reply:
RE: storing in DB—Gotcha, makes sense.  No need to change, just wanted to make sure I was thinking about that correctly for future reference.

RE: seeds—really appreciate you adding that "how to" in the PR description.

I know this is a bit "non-standard", but given we've got all the tools in the DB and need parity between dev/prod, what if the instructions added a sample migration for creating a new tool?  That way, a contributor could simply write a migration that could be run in both dev or prod for the tool?

```rb
class AddSomeNewTool < ActiveRecord::Migration[8.0]
  def up
    Tool.create!(
      name: "Stock Price Calculator",
      slug: "stock-price-calculator",
      intro: "Calculate and analyze stock prices with ease.",
      description: "Our Stock Price Calculator is a powerful tool designed to help investors and financial analysts quickly compute and evaluate stock prices.",
      content: "Use this tool to:\n\n- Calculate current stock values\n- Analyze historical price trends\n- Compare multiple stocks\n- Estimate future stock prices based on various metrics",
      category_slug: "finance",
      icon: "calculator",
      meta_image_url: "https://example.com/images/stock-calculator.jpg"
    )
  end

  def down
    Tool.find_by(slug: "stock-price-calculator")&.destroy
  end
end
```

Reply:
I'm not strongly opposed to that, but I think that'd be over-indexing on a pattern we'd actively like to get rid of.

We don't currently have parity between dev and prod, so instead of introducing a migration pattern to have parity going forward, I think ideally we'd fix at the root and just do away with db-backed tools in one fell swoop. Specially because I don't think it'd be a heavy lift.

BTW I just thought of a reason we might want the tools on the db. And that's if Avo provides WYSIWYG editing for the text. I'm not sure it does.

In the end, I'd make this decision based on who's gonna be editing the contents. Probably not the OSS devs, right? So that's another reason doing the migrations might not be as helpful — we wouldn't be able to merge + deploy without proofreading/editing.

As far as db-backed vs hardcoded, I'd say it depends on whether the person editing the contents is technical and comfortable with writing markdown in a heredoc, or whether they'd be more productive using WYSIWYG.

Reply:
Ahhh I think you might be right about Avo (I had not seen that when I was going through and reviewing).

Let's keep it as you've got it, I agree with the rationale here and you're almost certainly correct in saying that OSS contributors wouldn't be editing the _existence_ of tools anyways.